app/src/main/java/uk/ac/ebi/atlas/search/JsonGeneSearchController.java

```diff
@@ -262,7 +267,10 @@ public Set<String> getCellTypeBySearchTerm(@RequestParam MultiValueMap<String, S
             return ImmutableSet.of();
         }
 
-        return analyticsSearchService.searchCellType(geneIds.get());
+        var organismParts = requestParams.get("organismParts");
```

Comment:
I think the organism part endpoint code refactor is applicable here too. Code implementation is similar.

Reply:
I am sorry, but your comment is not clear to me.
Could you tell me with a bit more context what you mean, please?

Reply:
I have gone through the all articles you mentioned. I agree with you. Thanks.